crawlerr/services/strategies/web_strategy.py

```
@@ -0,0 +1,402 @@
+                    )
+
+                depth += 1
+
+        except Exception as e:
+            yield e
+
+    def _should_include_url(self, url: str, request: CrawlRequest) -> bool:
+        """Check if URL should be included in crawl."""
+        # Apply include patterns
+        if request.include_patterns:
+            if not any(pattern in url for pattern in request.include_patterns):
```

Ruff (0.12.2)

Use a single `if` statement instead of nested `if` statements

(SIM102)